ReactWindows/ReactNative.Shared/Common/AccessibilityHelper.cs

```diff
@@ -0,0 +1,229 @@
+            var parentAutomationPeer = uiElementAutomationPeer.Navigate(AutomationNavigationDirection.Parent) as AutomationPeer;
+            if (parentAutomationPeer is FrameworkElementAutomationPeer parentFEAutomationPeer)
+            {
+                if (parentFEAutomationPeer?.Owner != null)
+                {
+                    importantForAccessibilityAttached = GetImportantForAccessibilityAttached(parentFEAutomationPeer.Owner);
+                }
+            }
+            else if (parentAutomationPeer is ItemAutomationPeer parentIAutomationPeer)
+            {
+                if (parentIAutomationPeer?.Item is UIElement parentUIElement)
+                {
+                    importantForAccessibilityAttached = GetImportantForAccessibilityAttached(parentUIElement);
+                }
+            }
+
+            if (importantForAccessibilityAttached == ImportantForAccessibility.Yes ||
+                importantForAccessibilityAttached == ImportantForAccessibility.NoHideDescendants)
+            {
+                return true;
+            }
+            else
+            {
+                return IsHiddenElement(parentAutomationPeer);
+            }
+        }
+
+        /// <summary>
+        /// Recursively sets the value of AccessibilityView according ImportantForAccessibility setting
+        /// for children of uiElementAutomationPeer
+        /// </summary>
+        /// <param name="uiElementAutomationPeer"></param>
+        private static void SetChildrenAccessibilityViewFromImportantForAccessibility(AutomationPeer uiElementAutomationPeer)
+        {
+            if (uiElementAutomationPeer?.GetChildren() == null)
+            {
+                return;
+            }
+
+            foreach (AutomationPeer automationPeer in uiElementAutomationPeer.GetChildren())
+            {
+                UIElement childUIElement = null;
+                if (automationPeer is FrameworkElementAutomationPeer childFEAutomationPeer)
+                {
+                    if (childFEAutomationPeer?.Owner != null)
+                    {
+                        childUIElement = childFEAutomationPeer.Owner;
+                    }
+                }
+                else if (automationPeer is ItemAutomationPeer childIAutomationPeer)
+                {
+                    if (childIAutomationPeer?.Item is UIElement)
+                    {
+                        childUIElement = childIAutomationPeer?.Item as UIElement;
+                    }
+                }
+
+                if (childUIElement == null)
+                {
+                    SetChildrenAccessibilityViewFromImportantForAccessibility(automationPeer);
+                    continue;
+                }
+
+                var importantForAccessibilityAttached = GetImportantForAccessibilityAttached(childUIElement);
+                switch (importantForAccessibilityAttached)
```

Comment:
this switch seems to be identical to the one in SetImportantForAccessibility. Could it be be refactored into a separate method?

Reply:
Moved to a separate method.

---
In reply to: [163129031](https://github.com/Microsoft/react-native-windows/pull/1595#discussion_r163129031) [](ancestors = 163129031)